language/tools/move-mv-llvm-compiler/src/stackless/module_context.rs

```diff
@@ -622,4 +624,212 @@ impl<'mm, 'up> ModuleContext<'mm, 'up> {
             type_params,
         }
     }
+
+    fn generate_global_str_slice(&self, s: &str) -> llvm::Global {
+        let llcx = &self.llvm_cx;
+
+        // Create an LLVM global for the string.
+        let str_literal_init = llcx.const_int_array::<u8>(s.as_bytes()).as_const();
+        let str_literal = self
+            .llvm_module
+            .add_global2(str_literal_init.llvm_type(), "str_literal");
+        str_literal.set_constant();
+        str_literal.set_alignment(1);
+        str_literal.set_unnamed_addr();
+        str_literal.set_linkage(llvm::LLVMLinkage::LLVMPrivateLinkage);
+        str_literal.set_initializer(str_literal_init);
+
+        // Create an LLVM global for the slice, which is a struct with two fields:
+        // - pointer to the string literal
+        // - integer length of the string literal
+        let slice_len = s.len();
+        let slice_init = llcx.const_struct(&[
+            str_literal.ptr(),
+            llcx.const_int_array::<u8>(&slice_len.to_le_bytes())
+                .as_const(),
+        ]);
+        let slice = self
+            .llvm_module
+            .add_global2(slice_init.llvm_type(), "str_slice");
+        slice.set_constant();
+        slice.set_alignment(8);
+        slice.set_unnamed_addr();
+        slice.set_linkage(llvm::LLVMLinkage::LLVMPrivateLinkage);
+        slice.set_initializer(slice_init);
+
+        slice
+    }
+
+    fn emit_solana_entrypoint(&mut self) {
+        let entry_functions: Vec<_> = self
+            .env
+            .get_functions()
+            .filter(|fn_env| fn_env.is_entry())
+            .collect();
+        // Do not generate solana entrypoint if this is a script
+        if entry_functions.is_empty()
+            || entry_functions
+                .iter()
+                .any(|fn_env| fn_env.get_name_str() == "<SELF>")
```

Comment:
Should be able to use `fn_env.is_script()` instead.

Reply:
Thank you, I’ll use the is_script method instead.

Reply:
Unfortunately, this method returns true for entry functions also, not only for scripts. I need to exclude only scripts when solana entrypoint is generated.

https://github.com/solana-labs/move/blob/e310bc8c26b58e8d30a7a4267e6f645910141eb4/language/move-model/src/model.rs#L3306-L3310

Reply:
but I'll use is_script_module()